packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/describers.ts

```diff
@@ -53,6 +53,7 @@ class TypeNodeDescriber extends NodeDescriber {
   }
 
   public async gatherOutputLocations(): Promise<LocalComponent[]> {
+    await OrgBrowser.get().refreshAndExpand(this.node);
```

Comment:
This updates the list and expands the metadata type node before attempting to gather LocalComponents from the children.